File: e-elearning-main/h-learning-fe-admin/src/model/lession.ts
import { BaseModel } from "./base";
import { ChapterModel } from "./chapter";
import { CourseModel } from "./course";

export type LessionModel = BaseModel & {
    name: string
    description: string
    order: number
    chapterId: number
    courseId: number

    chapter?: ChapterModel
    course?: CourseModel
    // VideoLession *VideoLession `json:"videoLession" gorm:"foreignKey:LessionId"`
}

File: e-elearning-main/service/internal/entity/lession.go
package entity

import "gorm.io/gorm"

type Lession struct {
	gorm.Model
	Name        string `json:"name"`
	Description string `json:"description"`
	Order       int    `json:"order" gorm:"idx:idx_order_lession"`

	ChapterId uint `json:"chapterId" gorm:"idx:idx_order_lession"`
	CourseId  uint `json:"courseId" gorm:"idx:idx_order_lession"`

	Chapter      *Chapter      `json:"chapter" gorm:"foreignKey:ChapterId; constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
	Course       *Course       `json:"course" gorm:"foreignKey:CourseId; constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
	VideoLession *VideoLession `json:"videoLession" gorm:"foreignKey:LessionId"`
}
